lib/addons/helm-addon/kubectl-provider.ts

```diff
@@ -48,6 +48,15 @@ export interface HelmChartConfiguration {
      * Optional values for the helm chart. 
      */
     values?: Values
+
+    /**
+     * Indicate the helm chart provided uses dependency mode (https://helm.sh/docs/helm/helm_dependency/).
+     * Dependency mode is widely used in `aws-samples/ssp-eks-add-ons` repository, for example:
+     * https://github.com/aws-samples/ssp-eks-add-ons/blob/main/add-ons/appmesh-controller/Chart.yaml
+     * Dependency mode requires the chart values to be wrapped within the chart name.
+     * This value is only used when AddOns are deployed via GitOps way: `ArgoGitOpsFactory.enableGitOps()`
+     */
+    dependencyMode?: boolean
```

Comment:
I suspect there is a better way of doing it now with Argo. The examples that they provide still contain `dependency` for helm, but I can see that in [this](https://argo-cd.readthedocs.io/en/stable/user-guide/helm/#declarative) example `Application` CR was extended to speciify helm attributes for external repo. 
We can table it till upgrade to argo 5.x, an issue would help.

Reply:
I think Argo supports external repo, and there was some other reason [this repo](https://github.com/aws-samples/eks-blueprints-add-ons/) uses dependency mode, couldn't recall the reason though 👀

Reply:
No worries, let's get it in, and I will test with argo 5.x and if refactoring is needed I will apply.